homeassistant/components/technove/sensor.py

```diff
@@ -0,0 +1,164 @@
+        state_class=SensorStateClass.MEASUREMENT,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda station: station.info.current,
+    ),
+    TechnoVESensorEntityDescription(
+        key="energy_total",
+        translation_key="energy_total",
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda station: station.info.energy_total,
+    ),
+    TechnoVESensorEntityDescription(
+        key="energy_session",
+        translation_key="energy_session",
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        device_class=SensorDeviceClass.ENERGY,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda station: station.info.energy_session,
+    ),
+    TechnoVESensorEntityDescription(
+        key="rssi",
+        translation_key="rssi",
+        native_unit_of_measurement=SIGNAL_STRENGTH_DECIBELS_MILLIWATT,
+        device_class=SensorDeviceClass.SIGNAL_STRENGTH,
+        state_class=SensorStateClass.MEASUREMENT,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+        value_fn=lambda station: station.info.rssi,
+    ),
+    TechnoVESensorEntityDescription(
+        key="ssid",
+        translation_key="ssid",
+        icon="mdi:wifi",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+        value_fn=lambda station: station.info.network_ssid,
+    ),
+    TechnoVESensorEntityDescription(
+        key="status",
+        translation_key="status",
+        device_class=SensorDeviceClass.ENUM,
+        options=STATUS_TYPE,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        value_fn=lambda station: station.info.status.value,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the sensor platform."""
+    coordinator: TechnoVEDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+    async_add_entities(
+        TechnoVESensorEntity(coordinator, description) for description in SENSORS
+    )
+
+
+class TechnoVESensorEntity(TechnoVEEntity, SensorEntity):
+    """Defines a TechnoVE sensor entity."""
+
+    _attr_has_entity_name = True
+    entity_description: TechnoVESensorEntityDescription
+
+    def __init__(
+        self,
+        coordinator: TechnoVEDataUpdateCoordinator,
+        description: TechnoVESensorEntityDescription,
+    ) -> None:
+        """Initialize a TechnoVE sensor entity."""
+        super().__init__(coordinator=coordinator)
+        self.entity_description = description
+        self._attr_unique_id = f"{coordinator.data.info.mac_address}_{description.key}"
```

Comment:
I think you could move the unique id to the parent entity. Just create a constructor that can accept a coordinator and a key

Reply:
Sure, done.